ultravox/data/datasets.py

```diff
@@ -1206,37 +605,328 @@ class Range(SizedIterableDataset):
+    user_template=CONTINUATION_USER_TEMPLATE,
+    assistant_template=CONTINUATION_ASSISTANT_TEMPLATE,
+)
+CV_RU_CONT_CONFIG = DatasetConfig(
+    base="commonvoice-ru",
+    user_template=CONTINUATION_USER_TEMPLATE,
+    assistant_template=CONTINUATION_ASSISTANT_TEMPLATE,
+)
+
+LS_CLEAN_CONT_CONFIG = DatasetConfig(
+    base="librispeech-clean",
+    user_template=CONTINUATION_USER_TEMPLATE,
+    assistant_template=CONTINUATION_ASSISTANT_TEMPLATE,
+)
+
+LS_OTHER_CONT_CONFIG = DatasetConfig(
+    base="librispeech-other",
+    user_template=CONTINUATION_USER_TEMPLATE,
+    assistant_template=CONTINUATION_ASSISTANT_TEMPLATE,
+)
+
+PS_CLEAN_CONT_CONFIG = DatasetConfig(
+    base="peoplespeech",
+    user_template=CONTINUATION_USER_TEMPLATE,
+    assistant_template=CONTINUATION_ASSISTANT_TEMPLATE,
+)
+
+
+DATASET_MAP: Dict[str, DatasetConfig] = {
+    "boolq": BOOLQ_CONFIG,
+    "commonvoice": CV_BASE_CONFIG,
+    "commonvoice-en": CV_EN_CONFIG,
+    "commonvoice-ar": CV_AR_CONFIG,
+    "commonvoice-de": CV_DE_CONFIG,
+    "commonvoice-es": CV_ES_CONFIG,
+    "commonvoice-fr": CV_FR_CONFIG,
+    "commonvoice-it": CV_IT_CONFIG,
+    "commonvoice-ja": CV_JA_CONFIG,
+    "commonvoice-pt": CV_PT_CONFIG,
+    "commonvoice-ru": CV_RU_CONFIG,
+    "commonvoice-en-transcription": CV_EN_TRANS_CONFIG,
+    "commonvoice-ar-transcription": CV_AR_TRANS_CONFIG,
+    "commonvoice-de-transcription": CV_DE_TRANS_CONFIG,
+    "commonvoice-es-transcription": CV_ES_TRANS_CONFIG,
+    "commonvoice-fr-transcription": CV_FR_TRANS_CONFIG,
+    "commonvoice-it-transcription": CV_IT_TRANS_CONFIG,
+    "commonvoice-ja-transcription": CV_JA_TRANS_CONFIG,
+    "commonvoice-pt-transcription": CV_PT_TRANS_CONFIG,
+    "commonvoice-ru-transcription": CV_RU_TRANS_CONFIG,
+    "commonvoice-en-continuation": CV_EN_CONT_CONFIG,
+    "commonvoice-ar-continuation": CV_AR_CONT_CONFIG,
+    "commonvoice-de-continuation": CV_DE_CONT_CONFIG,
+    "commonvoice-es-continuation": CV_ES_CONT_CONFIG,
+    "commonvoice-fr-continuation": CV_FR_CONT_CONFIG,
+    "commonvoice-it-continuation": CV_IT_CONT_CONFIG,
+    "commonvoice-ja-continuation": CV_JA_CONT_CONFIG,
+    "commonvoice-pt-continuation": CV_PT_CONT_CONFIG,
+    "commonvoice-ru-continuation": CV_RU_CONT_CONFIG,
+    "gigaspeech": GS_XL_CONFIG,
+    "librispeech": LS_BASE_CONFIG,
+    "librispeech-clean": LS_CLEAN_CONFIG,
+    "librispeech-other": LS_OTHER_CONFIG,
+    "librispeech-clean-transcription": LS_CLEAN_TRANS_CONFIG,
+    "librispeech-other-transcription": LS_OTHER_TRANS_CONFIG,
+    "librispeech-clean-continuation": LS_CLEAN_CONT_CONFIG,
+    "librispeech-other-continuation": LS_OTHER_CONT_CONFIG,
+    "peoplespeech": PS_CLEAN_CONFIG,
+    "peoplespeech-transcription": PS_CLEAN_TRANS_CONFIG,
+    "peoplespeech-continuation": PS_CLEAN_CONT_CONFIG,
+    "voxpopuli": VP_EN_CONFIG,
+}
+
+
+def register_datasets(datasets: Dict[str, DatasetConfig]):
+    for name, config in datasets.items():
+        DATASET_MAP[name] = config
```

Comment:
should we do sanity check to avoid accidentally overriding datasets?

Reply:
good idea